devito/symbolics/extended_sympy.py

```diff
@@ -293,6 +294,40 @@ def __str__(self):
     __reduce_ex__ = Pickable.__reduce_ex__
 
 
+class DefFunction(Function, Pickable):
+
+    """
+    A definitely-defined sympy.Function, to work around:
+
+        https://github.com/sympy/sympy/issues/4297
+    """
+
+    def __new__(cls, name, arguments=None):
+        arguments = as_tuple(arguments)
+        obj = Function.__new__(cls, name, *arguments)
+        obj._name = name
+        obj._arguments = arguments
+        return obj
```

Comment:
That's...... :D But reading through the issue no really other way around it.

Reply:
yes I ditched 1 hour on this this morning than I gave up as it's just not worth it